data/lib/core/player.lua

```diff
@@ -534,3 +534,63 @@ function Player.sendBestiaryMilestoneReached(self, raceId)
 	msg:delete()
 	return true
 end
+
+function Player.sendHighscores(self, entries, params)
+	local msg = NetworkMessage()
+	msg:addByte(0xB1)
+
+	msg:addByte(params.action)
+
+	msg:addByte(0x01)
+	msg:addString(configManager.getString(configKeys.SERVER_NAME))
+	msg:addString(params.world)
+
+	msg:addByte(params.worldType)
+	msg:addByte(params.battlEye)
+
+	local vocations = Game.getUnpromotedVocations()
+	msg:addByte(#vocations + 1)
+
+	msg:addU32(0xFFFFFFFF)
+	msg:addString("All vocations")
+
+	for _, vocation in ipairs(vocations) do
+		msg:addU32(vocation:getId())
+		msg:addString(vocation:getName())
+	end
+
+	msg:addU32(params.vocation)
+
+	msg:addByte(#HIGHSCORES_CATEGORIES)
+	for id, category in ipairs(HIGHSCORES_CATEGORIES) do
```

Comment:
Why does this one in particular use `ipairs` instead of `pairs`?

Reply:
For this particular case it is fine to use ipairs, the difference is in nanoseconds. My apolgies to ranisalt for making him doubt about this. The only advantage of pairs is that when iterating through tables with nil we can iterate over the `nil` values.